src/bilder/images/edxapp/templates/vector/edxapp.yaml

```diff
@@ -19,19 +20,21 @@ sources:
 
 transforms:
 
-  edxapp_logs:
-    inputs:
-    - edxapp
-    type: filter
-    condition: match, err = contains(.message, "[service_variant=")
-
   edxapp_labeled_logs:
     inputs:
-    - edxapp_logs
+    - edxapp
     type: remap
     source: |
-      .labels = ["edxapp", "${ENVIRONMENT}"]
-      .environment = "${ENVIRONMENT}"
+      event, err = parse_json(.message)
+      if event != null {
+        . = merge!(., event)
+        .@timestamp = .timestamp
+        .labels = ["edxapp", "${ENVIRONMENT}"]
+        .application = "edxapp"
+        .environment = "${ENVIRONMENT}"
+      } else {
+        .malformed = true
```

Comment:
should this be documented in an informal schema, or otherwise made obvious, in case of a change and sudden ingress of malformed logs into ES?

Reply:
I think the general concept is good tho, and every log can and should have an 'event' tag in the json

Reply:
Definitely agree that we need to establish a core schema for log messages and labeling. The `event, err` piece is just arbitrary temporary variables that are used to store the results of the `parse_json` function call. The contents of `event` are just whatever attributes are contained in the JSON message.